Sources/MongoSwift/BSON/Document.swift

```diff
@@ -157,17 +157,51 @@ extension Document {
 
                 // otherwise, we just create a new document and replace this key
             } else {
-                // TODO: SWIFT-224: use va_list variant of bson_copy_to_excluding to improve performance
-                var newSelf = Document()
+                guard let iter = DocumentIterator(forDocument: self) else {
+                    throw InternalError(message: "BSON buffer is unexpectedly too small (< 5 bytes)")
```

Comment:
similarly to my comment below, if this error came from somewhere else, consider introducing a reusable error with this message.

Reply:
Done.